src/Model/Asset/AssetOptions.php

```diff
@@ -65,4 +78,54 @@ public function setExtensions(array $extensions): self
 
         return $this;
     }
+
+    public function toArray(): array
+    {
+        return [
+            'max_supply' => $this->getMaxSupply(),
+            'core_exchange_rate' => $this->getExchangeRate()->toArray(),
+            'is_exchangeable' => $this->isExchangeable(),
+            'extensions' => array_map(function ($extension) {return [1, $extension];}, $this->getExtensions())
+        ];
+    }
+
+    public function toBytes(): string
+    {
+        return implode('', [
+            str_pad(Math::gmpDecHex(Math::reverseBytesLong($this->getMaxSupply())), 16, '0', STR_PAD_LEFT),
+            $this->getExchangeRate()->toBytes(),
+            $this->isExchangeable() ? '01' : '00',
+            str_pad(count($this->getExtensions()), 2, '0', STR_PAD_LEFT),
+            // TODO: isFixedMaxSupply
+//            '01'
```

Comment:
there are multiple unresolved todos like this one, please go through them

Reply:
Resolved in #25.